Assets/BossRoom/Scripts/Client/UI/ConnectionStatusMessageUIManager.cs

```diff
@@ -9,12 +9,16 @@ namespace Unity.Multiplayer.Samples.BossRoom.Visual
     /// </summary>
     public class ConnectionStatusMessageUIManager : MonoBehaviour
     {
-        IDisposable m_Subscriptions;
+        DisposableGroup m_Subscriptions;
+
+        PopupPanel m_CurrentReconnectPopup;
 
         [Inject]
-        void InjectDependencies(ISubscriber<ConnectStatus> connectStatusSub)
+        void InjectDependencies(ISubscriber<ConnectStatus> connectStatusSub, ISubscriber<ReconnectMessage> reconnectMessageSub)
         {
-            m_Subscriptions = connectStatusSub.Subscribe(OnConnectStatus);
+            m_Subscriptions = new DisposableGroup();
+            m_Subscriptions.Add(connectStatusSub.Subscribe(OnConnectStatus));
```

Comment:
why do we have both connect status and reconnect message?
Might be worth it to add a quick comment on this

Reply:
nvm, just saw the extra fields for the reconnect message. @pdeschain might be worth it to have some form of inheritance for messages? Would a 3rd party library have this already built in? (I wouldn't add this to our own)